File: electron/src/renderer/lib/api/minutes.ts
// ============================================
// MINUTES API
// API functions for meeting minutes operations
// ============================================

import api from '../apiClient';

export interface MeetingMinutes {
  id: string;
  meeting_id: string;
  version: number;
  minutes_text?: string;
  minutes_html?: string;
  minutes_markdown?: string;
  executive_summary?: string;
  minutes_doc_url?: string;
  status: 'draft' | 'reviewed' | 'approved' | 'distributed';
  generated_at: string;
  edited_by?: string;
  edited_at?: string;
  approved_by?: string;
  approved_at?: string;
}

export interface MeetingMinutesList {
  minutes: MeetingMinutes[];
  total: number;
}

export interface GenerateMinutesRequest {
  meeting_id: string;
  template_id?: string; // Template ID to use for generation
  include_transcript?: boolean;
  include_actions?: boolean;
  include_decisions?: boolean;
  include_risks?: boolean;
  format?: 'markdown' | 'html' | 'text';
}

export interface DistributeMinutesRequest {
  minutes_id: string;
  meeting_id: string;
  channels?: string[]; // email, teams, sharepoint
  recipients?: string[]; // user_ids, undefined = all participants
}

export interface DistributionLog {
  id: string;
  minutes_id: string;
  meeting_id: string;
  user_id?: string;
  channel: string;
  recipient_email?: string;
  sent_at: string;
  status: string;
  error_message?: string;
}

export interface DistributionLogList {
  logs: DistributionLog[];
  total: number;
}

const ENDPOINT = '/minutes';

export const minutesApi = {
  /**
   * List all minutes versions for a meeting
   */
  list: async (meetingId: string): Promise<MeetingMinutesList> => {
    return api.get<MeetingMinutesList>(`${ENDPOINT}/${meetingId}`);
  },

  /**
   * Get the latest minutes for a meeting
   */
  getLatest: async (meetingId: string): Promise<MeetingMinutes | null> => {
    const response = await api.get<{ meeting_id: string; minutes: MeetingMinutes | null; message?: string }>(
      `${ENDPOINT}/${meetingId}/latest`
    );
    return response.minutes;
  },

  /**
   * Generate minutes using AI
   */
  generate: async (request: GenerateMinutesRequest): Promise<MeetingMinutes> => {
    return api.post<MeetingMinutes>(`${ENDPOINT}/generate`, request);
  },

  /**
   * Update minutes
   */
  update: async (minutesId: string, data: Partial<MeetingMinutes>): Promise<MeetingMinutes> => {
    return api.put<MeetingMinutes>(`${ENDPOINT}/${minutesId}`, data);
  },

  /**
   * Approve minutes
   */
  approve: async (minutesId: string, approvedBy: string): Promise<MeetingMinutes> => {
    return api.post<MeetingMinutes>(`${ENDPOINT}/${minutesId}/approve?approved_by=${encodeURIComponent(approvedBy)}`, null);
  },

  /**
   * Get distribution logs for a meeting
   */
  getDistributionLogs: async (meetingId: string): Promise<DistributionLogList> => {
    return api.get<DistributionLogList>(`${ENDPOINT}/${meetingId}/distribution`);
  },

  /**
   * Distribute minutes to participants
   */
  distribute: async (request: DistributeMinutesRequest): Promise<{
    status: string;
    distributed_to: number;
    channels: string[];
    logs: DistributionLog[];
  }> => {
    return api.post(`${ENDPOINT}/distribute`, request);
  },
};

export default minutesApi;

File: backend/app/schemas/minutes.py
"""
Meeting Minutes Schemas
"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class MeetingMinutesBase(BaseModel):
    version: int = 1
    minutes_text: Optional[str] = None
    minutes_html: Optional[str] = None
    minutes_markdown: Optional[str] = None
    executive_summary: Optional[str] = None
    status: str = "draft"  # draft / reviewed / approved / distributed


class MeetingMinutesCreate(MeetingMinutesBase):
    meeting_id: str


class MeetingMinutesUpdate(BaseModel):
    minutes_text: Optional[str] = None
    minutes_html: Optional[str] = None
    minutes_markdown: Optional[str] = None
    executive_summary: Optional[str] = None
    status: Optional[str] = None


class MeetingMinutesResponse(MeetingMinutesBase):
    id: str
    meeting_id: str
    minutes_doc_url: Optional[str] = None
    generated_at: datetime
    edited_by: Optional[str] = None
    edited_at: Optional[datetime] = None
    approved_by: Optional[str] = None
    approved_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class MeetingMinutesList(BaseModel):
    minutes: List[MeetingMinutesResponse]
    total: int


# ============================================
# Distribution Log Schemas
# ============================================

class DistributionLogBase(BaseModel):
    channel: str  # email / teams / sharepoint
    recipient_email: Optional[str] = None
    status: str = "sent"  # sent / delivered / read / failed


class DistributionLogCreate(DistributionLogBase):
    minutes_id: str
    meeting_id: str
    user_id: Optional[str] = None


class DistributionLogResponse(DistributionLogBase):
    id: str
    minutes_id: str
    meeting_id: str
    user_id: Optional[str] = None
    sent_at: datetime
    error_message: Optional[str] = None

    class Config:
        from_attributes = True


class DistributionLogList(BaseModel):
    logs: List[DistributionLogResponse]
    total: int


# ============================================
# Generate Minutes Request
# ============================================

class GenerateMinutesRequest(BaseModel):
    meeting_id: str
    template_id: Optional[str] = None  # Template ID to use for generation
    include_transcript: bool = True
    include_actions: bool = True
    include_decisions: bool = True
    include_risks: bool = True
    format: str = "markdown"  # markdown / html / text


class DistributeMinutesRequest(BaseModel):
    minutes_id: str
    meeting_id: str
    channels: List[str] = ["email"]  # email / teams / sharepoint
    recipients: Optional[List[str]] = None  # user_ids, None = all participants

